gluten-iceberg/src/main/scala/org/apache/iceberg/spark/source/GlutenIcebergSourceUtil.scala

```diff
@@ -94,15 +97,24 @@ object GlutenIcebergSourceUtil {
       throw new GlutenNotSupportException("Only support iceberg SparkBatchQueryScan.")
   }
 
-  def getPartitionSchema(sparkScan: Scan): StructType = sparkScan match {
+  def getReadPartitionSchema(sparkScan: Scan): StructType = sparkScan match {
```

Comment:
Maybe rename to `getReferredPartitionSchema` to indicates that we only retrieve the partition schemas referred  by the query?

Reply:
It just imitates the style of spark, see `Scan.readSchema`